contracts/sg-ics721/src/testing/integration_tests.rs

```diff
@@ -0,0 +1,1564 @@
+    // test case: receive nft from sg721-base
+    {
+        let mut test = Test::new(false, None, sg721_base_contract());
+        // mint and escrowed/owned by ics721
+        let token_id = test.execute_cw721_mint(test.ics721.clone()).unwrap();
+
+        let res = test
+            .app
+            .execute_contract(
+                test.cw721.clone(),
+                test.ics721.clone(),
+                &ExecuteMsg::ReceiveNft(cw721::Cw721ReceiveMsg {
+                    sender: test.minter.to_string(),
+                    token_id: token_id.clone(),
+                    msg: to_binary(&IbcOutgoingMsg {
+                        receiver: "mr-t".to_string(),
+                        channel_id: "channel-0".to_string(),
+                        timeout: IbcTimeout::with_block(IbcTimeoutBlock {
+                            revision: 0,
+                            height: 10,
+                        }),
+                        memo: None,
+                    })
+                    .unwrap(),
+                }),
+                &[],
+            )
+            .unwrap();
+        let event = res.events.into_iter().find(|e| e.ty == "wasm").unwrap();
+        let class_data_attribute = event
+            .attributes
+            .into_iter()
+            .find(|a| a.key == "class_data")
+            .unwrap();
+        let expected_contract_info: cosmwasm_std::ContractInfoResponse = from_binary(
+            &to_binary(&ContractInfoResponse {
+                code_id: test.cw721_id,
+                creator: test.minter.to_string(),
+                admin: None,
+                pinned: false,
+                ibc_port: None,
+            })
+            .unwrap(),
+        )
+        .unwrap();
+        assert_eq!(
+            class_data_attribute.value,
+            format!(
+                "{:?}",
+                // outbound, transfer from SG to another chain
+                SgCollectionData {
+                    owner: Some(test.minter.to_string()),
+                    contract_info: expected_contract_info,
+                    name: "name".to_string(),
+                    symbol: "symbol".to_string(),
+                    num_tokens: 1,
+                    collection_info: CollectionInfoResponse {
+                        creator: test.ics721.to_string(),
+                        description: "".to_string(),
+                        image: "https://arkprotocol.io".to_string(),
+                        external_link: None,
+                        explicit_content: None,
+                        start_trading_time: None,
+                        royalty_info: None,
+                    },
+                }
+            )
+        );
+    }
+    // test case: receive nft from old/v240 sg721-base
+    {
+        let mut test = Test::new(false, None, sg721_v240_base_contract());
+        // mint and escrowed/owned by ics721
+        let token_id = test.execute_cw721_mint(test.ics721.clone()).unwrap();
```

Comment:
Test outbound from old sg721base v2.4.0 contract (containing minter, not ownership).